src/components/EpicSelect.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <EpicInputItem :label="label" :input-id="id">
+    <select :id="id" @input="handleSelectInput" :value="selectedId">
+      <option
+        v-for="option in options"
+        :key="option.id"
+        :value="option.id"
+        :selected="selectedId === option.id"
+        >{{ option.display }}</option
+      >
+    </select>
+  </EpicInputItem>
+</template>
+
+<script lang="ts">
+import EpicInputItem from '@components/EpicInputItem'
+import { ReactiveBase } from '@helper/ReactiveBase'
+import { computed } from '@vue/reactivity'
+import { generate } from 'shortid'
+import { defineComponent, PropType } from 'vue'
+
+export class EpicSelectOption extends ReactiveBase {
+  /**
+   * @param value
+   * @param id If no id is present the select will create on using the shortId package
+   * @param display if no display is specified and no slot is provided the value will be interpreted as string
+   */
+  constructor(
+    public readonly value: unknown,
+    public readonly id: string = generate(),
+    public readonly display: string = String(value)
+  ) {
+    super()
+  }
+}
+
+export default defineComponent({
+  name: 'EpicSelect',
+  components: {
+    EpicInputItem,
+  },
+  props: {
+    options: {
+      type: Array as PropType<EpicSelectOption[]>,
+      required: true,
+    },
+    selectedItem: {
+      default() {
+        return null
+      },
+    },
+    label: {
+      type: String,
+      required: true,
+    },
+  },
+  setup(
+    props: { options: EpicSelectOption[]; selectedItem: unknown },
+    context
+  ) {
+    function handleSelectInput($event: InputEvent) {
+      const value = props.options.find((x) => x.id === $event.target.value)
+        ?.value
+
+      context.emit('update:selected-item', value)
+    }
+
+    const selectedId = computed(
+      () => props.options.find((x) => x.value === props.selectedItem)?.id
```

Comment:
Damit gehen wir davon aus, dass nicht nur die id, sondern auch die value eindeutig ist, oder?

Abgesehen davon würde das vermutlich Probleme machen, wenn die Value ein Objekt ist.

Wie wäre es, wenn wir grundsätzlich davon ausgehen, dass einer SelectOption eine ID von außen mitgegeben wird und man dann hier einfach props.selectedItem.id zurückgibt? Das SelectedItem müsste dann natürlich die komplette SelectOption sein und die müssten wir auch bei handleSelectInput zurückgeben, anstatt nur die value.

Als Fallback könnten wir auch SelectOptions als einfachen String anstatt einem Objekt mit id, value, etc. erwarten und dann von außen ein normales Array reingeben. In dem Fall wäre der String sowohl value, als auch id. Solange wir das EpicSelect nur für Strings verwenden würde uns das sogar reichen.

Reply:
Ja die value muss eindeutig sein. Es geht mit objekten (zumindest mit denen ichs getestet hab 😂)
Was ich vermeiden wollte ist das man immer die EpicSelectOption als selected item übergeben muss weil dann kann das select nicht mehr mit vmodel verwendet werden

Reply:
Hm, verstehe 🤔 Evtl. finden wir da künftig mal noch ne bessere Lösung. Aber solangs funzt 😄